### PR TITLE
CSV Import Add Members

### LINKED ISSUE #269: Batch Upload Contributors by CSV
**Current Behavior:** Admins can only enter new contributors manually via the Admin Page.

**Desired Behavior:** Admins can download a Template CSV, which contains any contributors and settings available in the circle. They can make changes and upload it again with new contributors populated or other changes.  This should not effect any current GIVE totals, etc.

### PR DESCRIPTION
**Pull request by CryptoGraffe**
## Motivation and Context

To allow for CSV import of new members, one of our oldest tickets and most requested features. Fixes #269 

Merging on this is blocked until #1270 is merged, or this can merge into that branch. 

## Description
* CSV Import for new members leveraging the existing bulk add members UI/impl for heavy lifting. 
* Refactoring and cleaning up some patterns in the other add member flows, reducing component complexity
* Adds a reusable component for drag and drop file upload

## Test and Deployment Plan

Test all the add member workflows including CSV with and without error states. Thorough testing is required because I refactored quite a bit. 

I tested a lot.

### CODE REVIEW

File: src/pages/AddMembersPage/AddMembersPage.tsx [hunk 1]
```diff
@@ -119,84 +139,67 @@ const AddMembersContents = ({
...
+              </Text>
+              <MagicLink magicLink={magicLink} />
+            </Box>
+          )}
+          {currentTab === Tab.CSV && (
+            <Box>
+              <Text css={{ pb: '$lg', pt: '$sm' }} size={'large'}>
+                Please import a .CSV file with only these two columns: Name,
+                Address. &nbsp;
+                <Text size={'small'}>
```
Comment (levity) [inline src/pages/AddMembersPage/AddMembersPage.tsx:184]: this should be the same size as the rest of the text

File: src/pages/AddMembersPage/AddMembersPage.tsx [hunk 2]
```diff
@@ -119,84 +139,67 @@ const AddMembersContents = ({
...
-            <Text css={{ pb: '$lg', pt: '$sm' }} size={'large'}>
-              Add new members by sharing a magic link.
-            </Text>
-            <ConstrainedBox css={{ mb: '$md' }}>
-              <Flex css={{ alignItems: 'center', mb: '$xs' }}>
-                <Text variant="label">
-                  Magic Circle Link
-                  {/*<ApeInfoTooltip>TODO: FILL THIS OUT</ApeInfoTooltip>*/}
+        <Box
+          css={{
```
Comment (levity) [inline src/pages/AddMembersPage/AddMembersPage.tsx:151]: i think it looks great when there's no extra space in the right side of the box :)

File: public/resources/example.csv [hunk 3]
```diff
@@ -0,0 +1,3 @@
+"Name","Address"
+"Panda Bear", "0xDc19f74a09A732762Ca84CB5B6a3426fd5aE799E"
```
Comment (levity) [inline public/resources/example.csv:2]:
```suggestion
"Panda Bear","0xDc19f74a09A732762Ca84CB5B6a3426fd5aE799E"
```

**Review by levity — CHANGES_REQUESTED** [review]

**Review by zashton — DISMISSED** [review]

File: src/pages/AddMembersPage/AddMembersPage.tsx [hunk 4]
```diff
@@ -119,86 +139,70 @@ const AddMembersContents = ({
...
+              <Text css={{ pb: '$lg', pt: '$sm' }} size={'large'}>
+                Add new members by sharing a magic link.
+              </Text>
+              <MagicLink magicLink={magicLink} />
+            </Box>
+          )}
+          {currentTab === Tab.CSV && (
+            <Box>
+              <Text css={{ pb: '$lg', pt: '$sm' }} size={'large'}>
+                Please import a .CSV file with only these two columns: Name,
```
Comment (zashton) [inline src/pages/AddMembersPage/AddMembersPage.tsx:183]: Some common fields that users usually want to set through import are `role` admin status, `starting_tokens`,`fixed_non_receiver`, `non_giver`

File: src/pages/AddMembersPage/NewMemberList.tsx [hunk 5]
```diff
@@ -149,8 +155,13 @@ const NewMemberList = ({
     }
   };
 
+  useEffect(() => {
+    // do initial form validation to validate the preloaded values (from csv or other import)
+    trigger().then();
```
[the remark this answers is not in the document]
Comment (CryptoGraffe) [reply src/pages/AddMembersPage/NewMemberList.tsx:182]: ended up keeping this cuz it makes my IDE happy

File: src/pages/AddMembersPage/AddMembersPage.tsx (hunk 2 above)
Comment (CryptoGraffe) [reply src/pages/AddMembersPage/AddMembersPage.tsx]: done

File: public/resources/example.csv (hunk 3 above)
Comment (CryptoGraffe) [reply public/resources/example.csv]: done

File: src/pages/AddMembersPage/AddMembersPage.tsx (hunk 4 above)
Comment (CryptoGraffe) [reply src/pages/AddMembersPage/AddMembersPage.tsx:183]: This was discussed but we are keeping MVP for simplicity right now, that can come as follow-on work

File: src/pages/AddMembersPage/AddMembersPage.tsx (hunk 1 above)
Comment (CryptoGraffe) [reply src/pages/AddMembersPage/AddMembersPage.tsx:185]: ok i changed it back

File: src/pages/AddMembersPage/MagicLink.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    <Box>
+      <Flex css={{ alignItems: 'center', mb: '$xs' }}>
+        <Text variant="label">Magic Circle Link</Text>
+      </Flex>
+      <CopyCodeTextField value={magicLink} />
+
+      <Panel alert css={{ mt: '$xl', mb: '$md' }}>
+        <Flex>
+          <AlertSolid
+            color={'alert'}
```
[the remark this answers is not in the document]
Comment (CryptoGraffe) [reply src/pages/AddMembersPage/MagicLink.tsx]: done

**Review by levity — APPROVED** [review]

**Pull request merged**